compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -972,7 +972,6 @@ public void visit(BLangXMLElementLiteral xmlElementLiteral) {
 
     @Override
     public void visit(BLangXMLTextLiteral xmlTextLiteral) {
-        xmlTextLiteral.textFragments.forEach(this::rewriteExpr);
```

Comment:
Shall we similarly fix the other XML kinds also?

e.g., the following also fails.

```ballerina
public function main() {
    xml _ = xml `<Books>${xml `<Book><!--${let int y = 5 in y}--></Book>`}</Books>`;
}
```